src/test/mir-opt/unusual-item-types.rs

```diff
@@ -1,12 +1,18 @@
 // Test that we don't ICE when trying to dump MIR for unusual item types and
 // that we don't create filenames containing `<` and `>`
+// compile-flags: --emit mir
 
 struct A;
 
 impl A {
     const ASSOCIATED_CONSTANT: i32 = 2;
 }
 
+// See #59021
+pub enum Test {
+    X(usize),
+}
```

Comment:
Can you add the output mir below? It should be in a file with a name like `build/x86_64-unknown-linux-gnu/test/mir-opt/unusual-item-types/rustc/Test-X.mir_map.0.mir`

Reply:
Also add the following to verify that we won't ICE with struct-like variants.

```
pub enum Test2 {
    X { a: usize },
}
```

Reply:
:+1: Added the tests